File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        if(nums.length == 1) {
            return nums[0];
        }
        
        int cursum = 0;
        int maxnum = nums[0];
        
        for(int i = 0; i < nums.length; i++) {
            if(cursum < 0) {
                cursum = 0;
            }
            cursum+= nums[i];
            maxnum = Math.max(cursum,maxnum);
        }
        
        return maxnum;
        
    }
}

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        
        curMax = -math.inf
        curSum = 0
        
        for num in nums:
            curSum+= num
            
            curMax = max(curMax,curSum)
            
            if curSum < 0:
                curSum = 0
        
        return curMax